packages/twenty-server/src/engine/metadata-modules/field-metadata/services/field-metadata-enum-validation.service.ts

```diff
@@ -0,0 +1,188 @@
+      },
+    ];
+
+    validators.forEach((validator) =>
+      this.validatorRunner(sanitizedLabel, validator),
+    );
+  }
+
+  private validateMetadataOptionValue(sanitizedValue: string) {
+    const validators: Validator<string>[] = [
+      {
+        validator: (value) => !isDefined(value),
+        message: 'Option value is required',
+      },
+      {
+        validator: exceedsDatabaseIdentifierMaximumLength,
+        message: `Option value "${sanitizedValue}" exceeds 63 characters`,
+      },
+      {
+        validator: beneathDatabaseIdentifierMinimumLength,
+        message: `Option value "${sanitizedValue}" is beneath 1 character`,
+      },
+      {
+        validator: (value) => !isSnakeCaseString(value),
+        message: `Value must be in UPPER_CASE and follow snake_case "${sanitizedValue}"`,
+      },
+    ];
+
+    validators.forEach((validator) =>
+      this.validatorRunner(sanitizedValue, validator),
+    );
+  }
+
+  private validateDuplicates(options: FieldMetadataOptions) {
+    const seenOptionIds = new Set<FieldMetadataOptions[number]['id']>();
+    const seenOptionValues = new Set<FieldMetadataOptions[number]['value']>();
+    const seenOptionPositions = new Set<
+      FieldMetadataOptions[number]['position']
+    >();
+
+    for (const option of options) {
+      if (seenOptionIds.has(option.id)) {
+        throw new FieldMetadataException(
+          `Duplicated option id "${option.id}"`,
+          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+        );
+      }
+
+      if (seenOptionValues.has(option.value)) {
+        throw new FieldMetadataException(
+          `Duplicated option value "${option.value}"`,
+          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+        );
+      }
+
+      if (seenOptionPositions.has(option.position)) {
+        throw new FieldMetadataException(
+          `Duplicated option position "${option.position}"`,
+          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+        );
+      }
+
+      seenOptionIds.add(option.id);
+      seenOptionValues.add(option.value);
+      seenOptionPositions.add(option.position);
+    }
+  }
+
+  async validateMetadataOptions({
+    fieldMetadataInput,
+    fieldMetadataType,
+  }: {
+    fieldMetadataInput: CreateFieldInput | UpdateFieldInput;
+    fieldMetadataType: FieldMetadataType;
+  }) {
+    if (!isEnumFieldMetadataType(fieldMetadataType)) {
+      return;
+    }
+
+    const { options } = fieldMetadataInput;
+
+    if (!isDefined(options) || options.length === 0) {
+      throw new FieldMetadataException(
+        'Options are required for enum fields',
+        FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+      );
+    }
+
+    for (const option of options) {
+      this.validateMetadataOptionId(option.id);
+      this.validateMetadataOptionValue(option.value);
+      this.validateMetadataOptionLabel(option.label);
+    }
+
+    this.validateDuplicates(options);
+
+    if (isDefined(fieldMetadataInput.defaultValue)) {
```

Comment:
I would actually go the other way, field-metadata-enum should not care about defaultValue
but defaultValue should make sure it's part of the options

In this case, I think field-metadata-enum-validation.service.ts does not need to inject fieldMetadataValidationService and can be an util

Reply:
Interesting about to see what I can do about that